From 5c55f16d04ae645d1708c09a0e4f9d349e13e69e Mon Sep 17 00:00:00 2001
From: John Howard <howardjohn@google.com>
Date: Fri, 11 Feb 2022 08:31:20 -0800
Subject: istioctl: fix order dependant tests (#37045) (#37246)

* istioctl: fix order dependant tests

* Fix cni and pilot test as well

(cherry picked from commit b872e512c22166a3dc338f37d45f20728b37c48c)
---
 cni/pkg/plugin/plugin_test.go | 1 +
 istioctl/cmd/tag_test.go      | 1 +
 istioctl/cmd/workload.go      | 3 +++
 3 files changed, 5 insertions(+)

diff --git a/cni/pkg/plugin/plugin_test.go b/cni/pkg/plugin/plugin_test.go
index c84561f0e6..31c5dc85e7 100644
--- a/cni/pkg/plugin/plugin_test.go
+++ b/cni/pkg/plugin/plugin_test.go
@@ -433,6 +433,7 @@ func TestCmdAddInvalidK8sArgsKeyword(t *testing.T) {
 
 func TestCmdAddInvalidVersion(t *testing.T) {
 	defer resetGlobalTestVariables()
+	getKubePodInfo = mockgetK8sPodInfo
 	testCmdInvalidVersion(t, CmdAdd)
 }
 
diff --git a/istioctl/cmd/tag_test.go b/istioctl/cmd/tag_test.go
index 967cc32e27..79e901de83 100644
--- a/istioctl/cmd/tag_test.go
+++ b/istioctl/cmd/tag_test.go
@@ -145,6 +145,7 @@ func TestTagList(t *testing.T) {
 		t.Run(tc.name, func(t *testing.T) {
 			var out bytes.Buffer
 			client := fake.NewSimpleClientset(tc.webhooks.DeepCopyObject(), tc.namespaces.DeepCopyObject())
+			revArgs.output = jsonFormat
 			err := listTags(context.Background(), client, &out)
 			if tc.error == "" && err != nil {
 				t.Fatalf("expected no error, got %v", err)
diff --git a/istioctl/cmd/workload.go b/istioctl/cmd/workload.go
index 47b3ce35c3..5d03ac4b5a 100644
--- a/istioctl/cmd/workload.go
+++ b/istioctl/cmd/workload.go
@@ -540,6 +540,9 @@ func marshalWorkloadEntryPodPorts(p map[string]uint32) string {
 	if len(out) == 0 {
 		return ""
 	}
+	sort.Slice(out, func(i, j int) bool {
+		return out[i].Name < out[j].Name
+	})
 	str, err := json.Marshal(out)
 	if err != nil {
 		return ""
-- 
2.35.3

